samza-core/src/main/java/org/apache/samza/zk/ZkJobCoordinator.java

```diff
@@ -356,10 +356,12 @@ JobModel generateNewJobModel(List<ProcessorNode> processorNodes) {
     // If JobModel exists in zookeeper && cached JobModel version is unequal to JobModel version stored in zookeeper.
     if (zkJobModelVersion != null && !Objects.equals(cachedJobModelVersion, zkJobModelVersion)) {
```

Comment:
Looks like TestZkLocalApplicationRunner.java needs to also be updated, there are some failed tests.